src/glfw/window.cc

```diff
@@ -0,0 +1,233 @@
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <GLFW/glfw3.h>
+#include <spdlog/spdlog.h>
+
+#include "window/error.h"
+#include "window/window.h"
+
+using namespace wnd;
+
+namespace
+{
+
+void
+setWindowHints()
+{
+    glfwDefaultWindowHints();
+
+    // clang-format off
+    glfwWindowHint( GLFW_RESIZABLE,               GLFW_TRUE );
+    glfwWindowHint( GLFW_VISIBLE,                 GLFW_TRUE );
+    glfwWindowHint( GLFW_FOCUSED,                 GLFW_TRUE );
+    glfwWindowHint( GLFW_FLOATING,                GLFW_FALSE );
+    glfwWindowHint( GLFW_TRANSPARENT_FRAMEBUFFER, GLFW_FALSE );
+    glfwWindowHint( GLFW_FOCUS_ON_SHOW,           GLFW_TRUE );
+    glfwWindowHint( GLFW_DOUBLEBUFFER,            GLFW_TRUE );
+    glfwWindowHint( GLFW_REFRESH_RATE,            GLFW_DONT_CARE );
+    glfwWindowHint( GLFW_CLIENT_API,              GLFW_NO_API );
+    // clang-format on
+} // setWindowHints
+
+std::string
+getHandleAddressString( const Window* window )
+{
+    return std::to_string( reinterpret_cast<uint64_t>( window ) );
+} // getHandleAddressString
+
+void
+logGLFWaction(
+    std::string&& action,
+    std::string&& handleAddressString = {},
+    std::string&& additional = {} //
+)
+{
+    std::string output = "GLFW " + action + ".";
+
+    if ( !handleAddressString.empty() )
+    {
+        // clang-format off
+        output += "\n"
+                  "Handle = " + handleAddressString;
+        // clang-format on
+
+        if ( !additional.empty() )
+        {
+            output += ", " + additional + ".";
+        }
+    }
+
+    spdlog::info( output );
+} // logGLFWaction
```

Comment:
Use ```fmt::format``` for string formatting instead of overloaded string operators.
They lead to excessive copying.

```cpp
std::string output = fmt::format("GLFW {}.", action);
auto oit = std::back_inserter(output);
if ( !handleAddressString.empty() )
{
    fmt::format_to(oit, "\nHandle = {}", handleAddressString); 
        
    if ( !additional.empty() ) 
    {
        fmt::format_to(oit, ", additional.{}", additional)
    }
}
```

Reply:
This looks more clearly, agree.